File: Packets/src/main/java/me/ImSpooks/core/packets/collection/network/PacketRequestConnection.java
package me.ImSpooks.core.packets.collection.network;

import lombok.Getter;
import me.ImSpooks.core.enums.ClientType;
import me.ImSpooks.core.packets.init.Packet;
import me.ImSpooks.core.packets.init.channels.WrappedInputStream;
import me.ImSpooks.core.packets.init.channels.WrappedOutputStream;
import me.ImSpooks.core.packets.security.shared.SharedEncryption;

import java.io.IOException;
import java.util.ArrayList;

/**
 * Created by Nick on 01 okt. 2019.
 * Copyright © ImSpooks
 */
@Getter
public class PacketRequestConnection extends Packet {

    private String password;
    private long randomKey;
    private String clientName;
    private ClientType clientType;

    public PacketRequestConnection(String password, long randomKey, String clientName, ClientType clientType) {
        this.password = password;
        this.randomKey = randomKey;
        this.clientName = clientName;
        this.clientType = clientType;
    }

    public PacketRequestConnection() {
    }

    @Override
    public void send(WrappedOutputStream out) throws IOException {
        if (SharedEncryption.getEncryption() == null)
            throw new RuntimeException("Encryption was not yet initialized");

        String encrypted = SharedEncryption.getEncryption().encryptCollection(this.password, this.randomKey);
        out.writeString(encrypted);
        out.writeString(this.clientName);
        out.writeInt(this.clientType.getId());
    }

    @Override
    public void receive(WrappedInputStream in) throws IOException {
        if (SharedEncryption.getEncryption() == null)
            throw new RuntimeException("Encryption was not yet initialized");

        String encrypted = in.readString();

        ArrayList result = SharedEncryption.getEncryption().decryptCollection(encrypted);
        this.password = (String) result.get(0);
        this.randomKey = (Long) result.get(1);

        this.clientName = in.readString();
        this.clientType = ClientType.getFromId(in.readInt());
    }
}

File: ClientBridge/src/main/php/client/packets/collection/network/PacketRequestConnection.php
<?php
/**
 * Created by Nick on 21 okt. 2019.
 * Copyright © ImSpooks
 */

namespace client\packets\collection\network;


use client\packets\init\channels\WrappedInputStream;
use client\packets\init\channels\WrappedOutputStream;
use client\packets\init\Packet;
use client\packets\security\shared\SharedEncryption;
use RuntimeException;

class PacketRequestConnection extends Packet {

    /** @var string */ private $password;
    /** @var int */ private $randomKey;
    /** @var string */ private $clientName;
    /** @var int */ private $clientType;

    public function __construct(string $password = "", int $randomKey = 0, string $clientName = "", int $clientType = 0) {
        $this->password = $password;
        $this->randomKey = $randomKey;
        $this->clientName = $clientName;
        $this->clientType = $clientType;
    }

    public function send(WrappedOutputStream $out) {
        if (SharedEncryption::getEncryption() == null)
            throw new RuntimeException("Encryption was not yet initialized");

        $encrypted = SharedEncryption::getEncryption()->encrypt(json_encode([$this->password, $this->randomKey]));
        $out->writeString($encrypted);
        $out->writeString($this->clientName);
        $out->writeInt($this->clientType);
    }

    public function receive(WrappedInputStream $in) {
        if (SharedEncryption::getEncryption() == null)
            throw new RuntimeException("Encryption was not yet initialized");

        $result = json_decode(SharedEncryption::getEncryption()->decrypt($in->readString()));
        var_dump($result);

        $this->password = strval($result[0]);
        $this->randomKey = intval($result[1]);

        $this->clientName = $in->readString();
        $this->clientType = $in->readInt();
    }

    public function getPassword(): string {
        return $this->password;
    }

    public function getRandomKey(): int {
        return $this->randomKey;
    }

    public function getClientName(): string {
        return $this->clientName;
    }

    public function getClientType(): int {
        return $this->clientType;
    }


}